client/allocrunner/taskrunner/secrets_hook.go

```diff
@@ -165,7 +165,7 @@ func (h *secretsHook) Prestart(ctx context.Context, req *interfaces.TaskPrestart
 	return nil
 }
 
-func (h *secretsHook) buildSecretProviders(secretDir string) ([]SecretProvider, error) {
+func (h *secretsHook) buildSecretProviders(secretDir string, nomadToken string) ([]SecretProvider, error) {
```

Comment:
We mint separate WI tokens for Vault. We probably don't want to give the default token to CPI plugins either, as it's not going to have the appropriate audience for third-parties.